quetz/authentication/azuread.py

```diff
@@ -0,0 +1,76 @@
+import json
+from .oauth2 import OAuthAuthenticator
+
+
+class AzureADAuthenticator(OAuthAuthenticator):
+    """Use Microsoft Azure Active Directory account to authenticate users with Quetz. 
+
+    To enable add the following to the configuration file:
+
+    .. code::
+
+      [azuread]
+      client_id = "some-client-id-value"
+      client_secret = "a-client-secret"
+      tenant_id = "tenant-name or id"
+
+    You can obtain ``client_id`` and ``client_secret`` by registering your
+    application with AzureAD platfrom at this URL:
+    `https://portal.azure.com/#blade/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/RegisteredApps`
+
+    The ``tenant_id`` can either be a specific tenant's GUID identifier or one of three values:
+    ``common``
+    ``organizations``
+    ``consumers``
+    """
+
+    provider = "azuread"
+    scope = "openid profile email"
+
+    async def userinfo(self, request, token):
+        response = await self.client.get("https://graph.microsoft.com/oidc/userinfo", token=token)
+        profile = response.json()
+
+        github_profile = {
+            "id": profile["sub"],
+            "name": profile["name"],
+            "avatar_url": "",
+            "login": profile["email"],
+        }
```

Comment:
looks all good! You might be interested in my recent work around email addresses. It would be good to also return a `emails: [{
"email": ...,
"verified": true/false,
"primary": true/false
}...]` 

list. I guess that data will somehow be available through Azure as well?

Reply:
#402